Prox/Prox/Utilities/Logger.swift

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+enum LoggerLevel: Int {
+    case debug
+    case info
+    case warn
+    case error
+}
+
+class Logger {
+    private let tag: String
+
+    /// All log levels >= this level will be printed.
+    var level: LoggerLevel
+
+    /// Create a Logger instance that prints messages with the given tag.
+    /// Messages are filtered out if they are less than the given level.
+    init(tag: String, level: LoggerLevel) {
+        self.tag = tag
+        self.level = level
+    }
+
+    /// Create a Logger instance that prints messages with the given tag.
+    /// Assigns the level based on the current build channel.
+    convenience init(tag: String) {
+        let level: LoggerLevel
+        switch AppConstants.BuildChannel {
+        case .Debug:
+            level = .debug
+        case .Release:
+            level = .warn
+        default:
+            level = .info
+        }
+
+        self.init(tag: tag, level: level)
+    }
+
+    func debug(_ message: Any?) {
+        log(message, levelTag: "debug", level: .debug)
+    }
+
+    func info(_ message: Any?) {
+        log(message, levelTag: "info", level: .info)
+    }
+
+    func warn(_ message: Any?) {
+        log(message, levelTag: "warn", level: .warn)
+    }
+
+    func error(_ message: Any?) {
+        log(message, levelTag: "error", level: .error)
+    }
+
+    private func log(_ message: Any?, levelTag: String, level: LoggerLevel) {
+        guard level.rawValue >= self.level.rawValue else { return }
+        print("\(tag) [\(levelTag)] \(message ?? "nil")")
```

Comment:
Should we use `NSLog` so these show up on BB? My understanding is that `NSLog` does but `print` does not.